packages/next-swc/crates/core/tests/fixture/optimize_server_react/3/output.js

```diff
@@ -0,0 +1,16 @@
+import { useState } from 'react';
+export default function App({ x }) {
+    const state = 0, setState = ()=>null;
+    const state2 = ()=>0, setState2 = ()=>null;
```

Comment:
```javascript
const state2 = (()=>0)(), setState2 = ()=>null;
```
In the input the type of state2 would be `number` where after optimizing it turned into `() => number`

Reply:
Good catch! I'll make sure to have it fixed.